### PR TITLE
gpu: Build and Ship an GPU enabled Kernel

### PR DESCRIPTION
**Pull request by zvonkok**
With each release make sure we ship a GPU and TEE enabled kernel

Fixes: #6553

DependsOn: https://github.com/kata-containers/kata-containers/pull/6522/files#diff-1c4cd801df522f4a92edbfb0fea95364ed074a391ea47c284ddc078f512f7b6a

### CODE REVIEW

**Comment by zvonkok** [comment]
This kernel can run GPU workloads with SNP,TDX enabled or disabled.

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh [hunk 1]
```diff
@@ -203,7 +219,23 @@ install_dragonball_experimental_kernel() {
...
+		"kernel-gpu-snp" \
+		"-x snp -g nvidia"
+}
+
+#Install GPU and TDX enabled kernel asset
+install_kernel_gpu_tdx() {
+	install_kernel_helper \
+		"assets.kernel-tdx-experimental.version" \
+		"kernel-gpu-tdx" \
+		"-x tdx -g nvidia"
```
Comment (fidencio) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:238]: You'll need something like:
```
	local kernel_url="$(get_from_kata_deps assets.kernel-tdx-experimental.url)"

	install_kernel_helper \
		"assets.kernel-tdx-experimental.version" \
		"kernel-tdx-experimental" \
		"-x tdx -u ${kernel_url}"
```

As we pass the `${kernel_url}` here, yq won't be used from inside the nested container.

**Review by fidencio — CHANGES_REQUESTED** [review]
You'll also need this one:
```
diff --git a/tools/packaging/static-build/kernel/Dockerfile b/tools/packaging/static-build/kernel/Dockerfile
index 4ccf2c0df..34a74a5ee 100644
--- a/tools/packaging/static-build/kernel/Dockerfile
+++ b/tools/packaging/static-build/kernel/Dockerfile
@@ -18,6 +18,7 @@ RUN apt-get update && \
            iptables \
            kmod \
            libelf-dev \
+           libssl-dev \
            patch && \
     if [ "$(uname -m)" = "s390x" ]; then apt-get install -y --no-install-recommends libssl-dev; fi && \
     apt-get clean && rm -rf /var/lib/lists/
```

As building `nvidia` modules requires openssl libraries.

**Comment by zvonkok** [comment]
I screwed up this one opening a new PR: https://github.com/kata-containers/kata-containers/pull/6595

**Pull request closed without merging**